src/formatters/default_formatter.erl

```diff
@@ -489,7 +490,7 @@ lay_no_comments(Node, Ctxt) ->
             beside(D, lay_text_float("."));
         binary ->
             Ctxt1 = reset_prec(Ctxt),
-            Es = lay_items(erl_syntax:binary_fields(Node), Ctxt1, fun lay/2),
+            Es = lay_items(erl_syntax:binary_fields(Node), Ctxt1#ctxt{in = binary}, fun lay/2),
```

Comment:
I'm not entirely convinced that this will work properly. 
We need to be very careful when introducing context like this for nested expressions to format.
For instance, once inside a binary, you can also be "outside" of a binary again if you do stuff like this…

```erlang
p() ->
    <<begin
          X = rand:uniform(10),
          X + ?A_MACRO(X)
      end/integer>>.
```

Reply:
Parentheses encapsulates data, they will work with or without them. The test case has been updated and your example has been added. Please take a look.